packages/cli/src/lib/snapshot/snapshot.spec.ts

```diff
@@ -0,0 +1,83 @@
+jest.mock('../platform/authenticatedClient');
+jest.mock('fs');
+
+import {createReadStream, ReadStream} from 'fs';
+import {join} from 'path';
+import {Readable} from 'stream';
+import {mocked} from 'ts-jest/utils';
+import {AuthenticatedClient} from '../platform/authenticatedClient';
+import {SnapshotFactory} from './snapshotFactory';
+
+const mockedCreateReadStream = mocked(createReadStream);
+const mockedAuthenticatedClient = mocked(AuthenticatedClient);
+const mockedCreateSnapshotFromFile = jest.fn();
+
+const doMockReadStream = () => {
+  mockedCreateReadStream.mockImplementation(() => {
+    const buffer = Buffer.from('this is a tést');
+    const readable = new Readable();
+    readable._read = () => {};
+    readable.push(buffer);
+    readable.push(null);
+    return readable as unknown as ReadStream;
+  });
+};
+
+const doMockAuthenticatedClient = () => {
+  mockedAuthenticatedClient.mockImplementation(
+    () =>
+      ({
+        getClient: () =>
+          Promise.resolve({
+            resourceSnapshot: {createFromFile: mockedCreateSnapshotFromFile},
+          }),
+      } as unknown as AuthenticatedClient)
+  );
+};
```

Comment:
Looks like it still causes the same error. Now the `resourceSnapshot` mock is missing all its methods.

Reply:
Hm yeah, the `PlatformClient` was missing some stuff.
Double check but I think this would work tho
```suggestion
const doMockAuthenticatedClient = () =>
  mockedAuthenticatedClient.prototype.getClient.mockImplementation(() =>
    Promise.resolve<PlatformClient>({
      resourceSnapshot: {createFromFile: mockedCreateSnapshotFromFile},
    } as unknown as PlatformClient)
  );

```